### PR TITLE
Add hash value typedef.

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Add a typedef for the various hash value variables. Use of a typedef conveys some additional meaning.

**Motivation and Context**
I kept forgetting whether hash values were int64_t or uint64_t.

### CODE REVIEW

File: include/onnxruntime/core/framework/execution_provider.h [hunk 1]
```diff
@@ -286,11 +288,11 @@ class IExecutionProvider {
   // multiple sessions.
   class ModelMetadefIdGenerator {
    public:
-    int GenerateId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash);
+    int GenerateId(const onnxruntime::GraphViewer& graph_viewer, HashValue& model_hash);
 
    private:
-    std::unordered_map<uint64_t, int64_t> main_graph_hash_;  // map graph instance hash to model contents hash
-    std::unordered_map<int64_t, int> model_metadef_id_;      // current unique id for model
+    std::unordered_map<HashValue, HashValue> main_graph_hash_;  // map graph instance hash to model contents hash
```
Comment (edgchen1) [inline include/onnxruntime/core/framework/execution_provider.h:294]: note: this is actually changing from `unordered_map<uint64_t, int64_t>` to `unordered_map<uint64_t, uint64_t>`. as far as I know, that is ok.

**Comment by yuslepukhin** [comment]
Would it be cheaper to write a note to yourself ? :)

File: include/onnxruntime/core/common/hash_value.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cstdint>
+
+namespace onnxruntime {
+
+/** A computed hash value. */
+using HashValue = uint64_t;
```
Comment (pranavsharma) [inline include/onnxruntime/core/common/hash_value.h:11]: Would HashValue_t or HashValueType be better to convey that this is a type?

Comment (edgchen1) [reply include/onnxruntime/core/common/hash_value.h:11]: some other examples to consider for precedent:
https://github.com/microsoft/onnxruntime/blob/9d9ebd3b8536cc93fe8ac32acdffcd490c16c70c/include/onnxruntime/core/graph/basic_types.h#L27-L32

given a name like "NodeIndex", "HashValue" seemed reasonable to me

Comment (pranavsharma) [reply include/onnxruntime/core/common/hash_value.h:11]: Ok, looks like we're inconsistent given that we've ProviderType as well.

**Review by pranavsharma — APPROVED** [review]

File: include/onnxruntime/core/common/hash_value.h (hunk 2 above)
Comment (gwang-msft) [reply include/onnxruntime/core/common/hash_value.h:11]: Just curious, why not put this into the `basic_types.h`, for simplicity?

**Review by gwang-msft — APPROVED** [review]

File: include/onnxruntime/core/common/hash_value.h (hunk 2 above)
Comment (edgchen1) [reply include/onnxruntime/core/common/hash_value.h:11]: > Just curious, why not put this into the basic_types.h, for simplicity?

basic_types.h currently seems to be graph-specific. but perhaps I can add a more general one.

Comment (edgchen1) [reply include/onnxruntime/core/common/hash_value.h]: moved HashValue to core/common/basic_types.h

**Review by gwang-msft — APPROVED** [review]

**Pull request merged**